internal/util/logging/handler.go

```diff
@@ -140,6 +141,8 @@ func NewHandler(out io.Writer, opts *NewHandlerOpts) *Handler {
 		h = slog.NewTextHandler(out, stdOpts)
 	case "json":
 		h = slog.NewJSONHandler(out, stdOpts)
+	case "embeddable":
+		h = newEmbeddableHandler(opts, nil)
```

Comment:
This one is odd one out, it implicitly doesn't use `io.Writer` instead it uses `opts.Handler`. Could you add a comment for special handling of embedded handler please. Also checks like below helps set expected values.
```
must.BeZero(out)
must.NotBeZero(opts.Handler)
```